### ISSUE TITLE
Unexpected autocoplete behaviour

### ISSUE THREAD

**Issue by Kaarel** [nats-io/natscli#502]
Added the output of `nats --completion-script-bash` to my `.bashrc`

Works awesome! with some caveats :)

- nats r<tab><tab> - prints reply    request  restore  rtt ✅ 
- nats s<tab> - autocompletes to nats seal ❌ 
- nats st<tab><tab> - does nothing ❌ 
- nats str<tab><tab> - does nothing ❌ 
- nats stre<tab><tab> - autocompletes to nats stream ✅

**Comment by sylr** [comment]
With zsh for me it only autocompletes the first word and then is for the second word is propose the list of first words:
```
 ✘ sylvain $ nats stream <tab>
account    backup     bench      consumer   context    errors     events     governor   help       kv         latency    object     publish    reply      request    restore    rtt        schema     server     stream     subscribe
```

**Comment by sylr** [comment]
@ripienaar any chance someone could look into that ? it's pretty annoying.

**Comment by ripienaar** [comment]
Agree, quite annoying, but I havnt had time to investigate sorry, help would be appreciated since it's such a infrequently used feature.

The behavior is somewhere deep in the CLI library, I did see there's some hints I can do to help it out but havnt investigated actualy using them.

**Comment by philpennock** [comment]
The Synadia installer for client tools provides `https://get-nats.io/zsh.complete.nats` aka https://github.com/ConnectEverything/client-tools/blob/main/zsh.complete.nats 

That wrapper manages to invoke the CLI-bundled completion interfaces correctly enough that you don't see the issue.  But it's not auto-updated along with the client binary.  Still, it works today.  There's also a `zshrc` with some copiously-commented examples of things like setting up tab-completion for NATS_CONTEXT using the completion function.

Download `https://get-nats.io/zsh.complete.nats` to be called `_nats` somewhere in your `$fpath` list and restart zsh to get fresh completion and see if that helps?
